client/blocks/upe/upe-deferred-intent-creation/payment-elements.js

```diff
@@ -19,31 +25,96 @@ import { getBlocksConfiguration } from 'wcstripe/blocks/utils';
  *
  * @return {JSX.Element} Rendered Payment elements.
  */
-const PaymentElements = ( { api, ...props } ) => {
+const PaymentElements = ( {
+	api,
+	supportsDeferredIntent,
+	components: { LoadingMask },
+	...props
+} ) => {
+	const [ clientSecret, setClientSecret ] = useState( null );
+	const [ paymentIntentId, setPaymentIntentId ] = useState( null );
+	const [ hasRequestedIntent, setHasRequestedIntent ] = useState( false );
+
+	useEffect( () => {
+		if ( supportsDeferredIntent || hasRequestedIntent ) {
+			return;
+		}
+
+		async function createIntent() {
+			try {
+				const response = await api.createIntent(
+					getBlocksConfiguration()?.orderId,
+					props.paymentMethodId
```

Comment:
Is `paymentMethodId` passed only when deferred intent is not supported? If not, I think it'd look more consistent to destructure it as we're doing with other props.

Reply:
I agree. Fixed in ebc28b2aaee61f211f8b50d0f156123bccc3d548.